src/GlobalDiffEq.jl

```diff
@@ -25,6 +27,71 @@ function DiffEqBase.__solve(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase
     return sol
 end
 
-export GlobalRichardson
+struct GlobalAdjoint{A} <: GlobalDiffEqAlgorithm
+    alg::A
+end
+
+function DiffEqBase.__solve(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.AbstractDAEProblem},
+                            alg::GlobalAdjoint, args...;
+                            dt,kwargs...)
+
+    opt = Dict(kwargs)
+    otheropts = delete!(copy(opt), :dt)
+    tstops = get(opt, :tstops, range(prob.tspan[1], stop=prob.tspan[2], step=dt))
+    local sol
+
+    n = length(prob.u0)
+    k = 2 # 2 random vectors
+    C = 11/6 # for k = 2
+    E₂ = 2/π
+    Eₙ = vec_expt_full(n)
+    gtol_get = get(opt, :abstol, 1e-6) # target gtol is abstol? abstol cannot be zero?
+    z₁, z₂ = orth_vec(n)
+
+    function adjoint_loss_z₁(u0, p)
+        # g(u), reverse-mode adjoint
+        _prob = remake(prob, u0=u0, p=p)
+        _sol = solve(_prob, alg.alg; dt=dt, sensealg=DiffEqSensitivity.QuadratureAdjoint())
+        z₁' * _sol.u[end]
+    end
+
+    function adjoint_loss_z₂(u0, p)
+        # g(u), reverse-mode adjoint
+        _prob = remake(prob, u0=u0, p=p)
+        _sol = solve(_prob, alg.alg; dt=dt, sensealg=DiffEqSensitivity.QuadratureAdjoint())
+        z₂' * _sol.u[end]
+    end
+
+    # get sensitivities with respect to p at each t
+    # only iterate over tstops if p is a function of t
+    is_callable(f) = !isempty(methods(f))
+    if is_callable(prob.p)
```

Comment:
`p` should never be callable. What's going on here?

Reply:
I have revised the adjoint solver, so this step is no-longer used. However, I believe that p (the parameters) is callable if it is a function of t, as shown in the pendulum testing example. Currently, the DiffEqSensitivity.QuadratureAdjoint() function attempts to take the length of the parameters p and thus throws an error if they are a function. This is the cause of the test failing in my second commit. This error occurred in the previous version of the code as well, and this check was a failed attempt to solve it.